### ISSUE TITLE
chore: setting up static sharding fleet for Status

### ISSUE THREAD

**Issue by jm-clius** [waku-org/research#9]
## Background

This issue tracks the work necessary to set up a static sharding fleet for Status Communities. This forms part of the [10K users epic](https://github.com/waku-org/pm/issues/12).
It involves:
- setting the requirements for the fleet and creating an issue in https://github.com/status-im/infra-status
- enumerating responsibilities and transferring ownership to a new fleet owner: https://github.com/waku-org/research/issues/8

### Why is this issue not in an infra repo?

Some requirements need to be agreed upon, before opening an issue in the relevant infra repo.

## Requirements

I suggest the following configuration:
- Fleet of (at least) 10 nodes, split into two sub-fleets:
  - `status.sharding.store`: 5 nodes configured only with `relay` and `store`. These will be the store/historical message providers
  - `status.sharding.bootstrap`: 5 nodes configured with `relay`, `filter`, `lightpush`, `peer-exchange`. These are the main bootstrap nodes and also provide services to resource-restricted nodes.
- all nodes should be configured _only_ for the Status Internal CC Community static shards
- `status.sharding.store` should be configured with a single, shared PostgreSQL backend
- status-go nodes should preferably only use the bootstrap nodes to prime their discv5 routing tables, as minimal mechanism to limit unnecessary interaction with store nodes.
- no websockets configuration for now
- 1 node in `status.sharding.bootstrap` should be set up with trace-level message logs, in order to facilitate future end-to-end message tracing and debugging.

**Comment by jm-clius** [comment]
Some questions on the above:
- @richard-ramos can you take a look if this approach makes sense, especially splitting the fleet between the bootstrap nodes and the store nodes? Would it be possible/easier to e.g. set up two different DNS node trees - one for `store` and one for `bootstrapping`? The status-go nodes would then populate their store providers with one query and use another query to bootstrap connection to the network? The alternative is to simply use a single DNS list retrievable via a single query, and then use the already-existing capability differentiation to populate `store` nodes and bootstrap to the remaining nodes.
- @richard-ramos do we already have an idea of what specific shards we'd use for the first (internal CCs?) Status Community?
- @Ivansete-status we probably want to prioritise final postgresql deployment for `wakuv2.sharding` and use that as a blueprint to create something similar for `status.sharding.store`.

**Comment by richard-ramos** [comment]
`All nodes should be configured only for the Status Internal CC Community static shards` causes me some confusion - what should be the behavior for the app for both new users (not having joined a community), and the behavior for users that are part of the status community. Should new users utilize the `status.prod` fleet? and *only* if they join the status community, they should use this new fleet? if so, it means that status-go should somehow associate communities to fleets. Such behavior is currently not implemented.

Something else to take into account is that status-go currently behaves like this:
1. It uses DNSDiscovery to obtain the discv5 bootstrap nodes to discover relay peers and filter peers (if using light mode). 
2. The list of store nodes is hardcoded in status-go (although you can manually add nodes by invoking a rpc method).

We probably should discuss this with status team to design what makes sense for retrieving message history, since I assume that `status.sharding.store` will return messages only from the status shards, but then, 1:1 messages, group chats and other communities for the time being use the default pubsub topic. So some heuristic needs to be formulated in order to decide what store nodes to use to retrieve messages (hardcoded store node lists, or associating fleets to communities or TBD).

The same applies for peer discovery for shards, (which is an open item for status-go). Should status-go use `status.sharding.bootstrap` for discovering all peers regardless of the shards they belong to? or should we use both the nodes from status.prod fleet and `status.sharding.bootstrap` fleet.

**Comment by richard-ramos** [comment]
> do we already have an idea of what specific shards we'd use for the first (internal CCs?) Status Community?

Nope, but setting a shard cluster/index for a community is easy-sh. For now i imagine that setting up any shard index between 128 - 767 in cluster 16 and using that should be fine?

**Comment by alrevuelta** [comment]
@jm-clius Perhaps this should go somewhere else? Its more related to deployments and ofc there are some things to figure out, but I don't see this as a research problem?

**Comment by jm-clius** [comment]
> what should be the behavior for the app for both new users (not having joined a community), and the behavior for users that are part of the status community

I would say there shouldn't really be any complicated logic done her for specific fleets. The heuristic I suggest (for now):
- a Status node is subscribed to a couple of static shards via relay (it may have other static shards configured for light protocols only, e.g. certain control message shards)
- user running the app for the first time may have some default static shard subscriptions (if e.g. it needs 1:1 messages by default, there should be a static shard assigned for that and be part of the initial app subscriptions)
- user can join Status Community, which would add relay subscription to the appropriate static shard(s)
- when user opens the app, a DNS query is performed which returns a list of bootstrap nodes. The discovery protocol should filter only those nodes that serve the subscribed static shards the user is interested in. This could also apply to populating the store node table.
- user continues using the shared discovery layer (which will be across all fleets), but keep filtering for nodes that belong to the static shards the user is interested in (i.e. only connect to these).

From what I understand, status-go does not yet support filtering the subscribed shards in peer discovery? That would need to be implemented, but shouldn't for now affect how we deploy the fleet - namely a fleet specifically configured to serve (only) the Status community. We _could_ consider using the same fleet for the 1:1/group chat message static shards for now, though we'll likely split this off in future as well. The point is that afaics there would be no need for any fleet-specific configuration if we have a shared discovery layer and a Status node that can filter discovered peers on shard.

The assumption is that we'll launch first for some test communities and thereafter only for the Status Internal CC community. We can choose to expand support on this fleet for more shards to make the process simpler, but at no point should anything other than static shards be used (i.e. Status nodes shouldn't be subscribed to the default pubsub topic). We can also increment towards something more sophisticated here (e.g. use the same store nodes for all shards and keep hardcoding these in the interim).
